packages/venia-concept/src/components/MiniCart/__stories__/MiniCart.js

```diff
@@ -0,0 +1,165 @@
+    image: {
+        __typename: 'MediaGalleryEntry',
+        disabled: false,
+        file: '/v/a/va11-sg_main.jpg',
+        label: 'Main',
+        position: 1
+    },
+    item_id: 1,
+    name: 'A Non-Configurable Product',
+    options: [],
+    price: 49,
+    product_type: 'simple',
+    qty: 1,
+    quote_id: '1234',
+    sku: 'SKU'
+};
+const configurableItem = {
+    ...nonConfigurableItem,
+    __typename: 'ConfigurableProduct',
+    image: {
+        ...nonConfigurableItem.image,
+        file: '/v/t/vt11-rn_main_1.jpg'
+    },
+    name: 'A Configurable Product',
+    options: [
+        { value: 'Rain', label: 'Fashion Color' },
+        { value: 'M', label: 'Fashion Size' }
+    ],
+    product_type: 'configurable'
+};
+
+const baseProps = {
+    beginEditItem: noop,
+    cart: {
+        details: {
+            currency: {
+                quote_currency_code: 'USD'
+            },
+            items: [],
+            items_qty: 0
+        },
+        isEditingItem: false,
+        totals: {
+            items: [],
+            subtotal: 49
+        }
+    },
+    classes: defaultClasses,
+    closeDrawer: noop,
+    endEditItem: noop,
+    isCartEmpty: true,
+    isMiniCartMaskOpen: false,
+    isOpen: true,
+    removeItemFromCart: noop,
+    updateItemInCart: noop
+};
+
+const stories = storiesOf('MiniCart', module);
+
+stories.add('Loading', () => {
+    const props = {
+        ...baseProps,
+        cart: {
+            ...baseProps.cart,
+            isLoading: true
+        }
+    };
+
+    return (
+        <Adapter
+            apiBase={apiBase}
+            apollo={{ link: Adapter.apolloLink(apiBase) }}
+            store={store}
+        >
+            <MiniCart {...props} />
+        </Adapter>
+    );
+});
+
+stories.add('Empty Cart', () => {
+    const props = {
+        ...baseProps
+    };
+
+    return (
+        <Adapter
+            apiBase={apiBase}
+            apollo={{ link: Adapter.apolloLink(apiBase) }}
+            store={store}
+        >
+            <MiniCart {...props} />
+        </Adapter>
+    );
+});
+
+stories.add('Many Items', () => {
+    const props = {
+        ...baseProps,
+        cart: {
+            ...baseProps.cart,
+            details: {
+                ...baseProps.cart.details,
+                items: [
+                    configurableItem,
+                    nonConfigurableItem,
+                    configurableItem,
+                    nonConfigurableItem,
+                    configurableItem,
+                    nonConfigurableItem,
+                    configurableItem
+                ],
+                items_qty: 7
+            }
+        },
+        isCartEmpty: false
+    };
+
+    return (
+        <Adapter
+            apiBase={apiBase}
+            apollo={{ link: Adapter.apolloLink(apiBase) }}
+            store={store}
+        >
+            <MiniCart {...props} />
+        </Adapter>
+    );
+});
+
+stories.add('Editing', () => {
```

Comment:
How hard would it be to add a non-configurable and a configurable edit story?

Reply:
Here's the error I get:

![Screen Shot 2019-06-05 at 12 48 43 PM](https://user-images.githubusercontent.com/13182778/58978807-42f77780-8792-11e9-99cc-fed4772506aa.png)

`EditItem` runs a graphql query for configurable items, and it never gets past this line:

```js
    const payload = await apolloClient.query({
      query,
      variables
    });

    // never gets to here
    receiveResponse(payload);
```

Reply:
Yea I would have hoped you could just pass down the data and it would only need a mocked wrapper at the top level.